modules/communications/communications.py

```diff
@@ -0,0 +1,58 @@
+import time
+from modules.common.logger.modules import logger
+from modules.common.mavlink.modules.drone_odometry import DronePosition
+from modules.detection_in_world import DetectionInWorld
+from modules.flight_interface.local_global_conversion import drone_position_global_from_local
+
+
+class Communications:
+    """
+    """
+    __create_key = object()
+
+    @classmethod
+    def create(
+        cls,
+        local_logger: logger.Logger,
+    ) -> "tuple[bool, Communications | None]":
+        """
+        Logs data and forwards it.
+        """
+
+        return True, Communications(cls.__create_key, local_logger)
+
+    def __init__(
+        self,
+        class_private_create_key: object,
+        local_logger: logger.Logger,
+    ) -> None:
+        """
+        Private constructor, use create() method.
+        """
+        assert class_private_create_key is Communications.__create_key, "Use create() method"
+
+        self.__logger = local_logger
+
+    def run(
+        self, detections_in_world: list[DetectionInWorld], home_location: DronePosition
+    ) -> tuple[bool, list[DetectionInWorld] | None]:
+        for detection_in_world in detections_in_world:
+            if len(detection_in_world.centre) == 3:
+                result, detection_in_world_global = drone_position_global_from_local(
```

Comment:
Create a new function in the `local_global_conversion` library that returns a `DetectionInWorld` in global space after taking in home position and `DetectionInWorld` in local space.

If you want to be extra safe, you could create a new class solely for global locations (e.g. `LocationGlobal` ). Hmm, now that I think about it, surely that class must already exist somewhere (maybe in pathing repository)?